src/datepicker/bs-datepicker.component.ts

```diff
@@ -173,6 +173,10 @@ export class BsDatepickerDirective implements OnInit, OnDestroy, OnChanges {
         this.hide();
       })
     );
+
+    this._outsideScrollListener = this._renderer.listen('document', 'scroll', () => {
```

Comment:
Isn't this supposed to be in another PR? Also, this behavior should be disabled by default and this listener should only be created when corresponding input is set to true. input example `@Input() hideOnScroll = false`;